### PR TITLE
Enhancement/4692 gathering data

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue:

- #4692 

## Relevant technical choices

- The text color used in the Figma designs, #ccc does not pass accessibility standards. I've used #828286 for now to be compliant.
- The IB says to Adsense `ModuleOverviewWidget` component, however we don't have the `isGatheringData` selector for Adsense and so will not be implemented. [More details](https://github.com/google/site-kit-wp/issues/4696#issuecomment-1040355199).
- SB6 stories version has been added for `DashboardOverallPageMetricsWidget.js`
- For the SearchFunnel widget, there is another PR https://github.com/google/site-kit-wp/pull/4862 which caters for the Analytics part, when it's not connected or does not have any goals.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/GatheringDataNotice.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	);
+}
+
+GatheringDataNotice.propTypes = {
+	style: PropTypes.oneOf( [ 'small', 'default', 'overlay' ] ),
+};
+
+GatheringDataNotice.defaultProps = {
+	style: 'default',
+};
```
Comment (eugene-manuilov) [inline assets/js/components/GatheringDataNotice.js:45]: Default props on functional components will be deprecated. We need to use default values when we destruct the `props` parameter:

```diff
- function GatheringDataNotice( { style } ) {
+ function GatheringDataNotice( { style = 'default' } ) {
      ...
  }

  ...

- GatheringDataNotice.defaultProps = {
-     style: 'default',
- };
```

File: assets/js/components/GatheringDataNotice.stories.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * Internal dependencies
+ */
+import GatheringDataNotice from './GatheringDataNotice';
+
+const Template = ( args ) => <GatheringDataNotice { ...args } />;
+
+export const DefaultGatheringDataNotice = Template.bind( {} );
+DefaultGatheringDataNotice.storyName = 'Default GatheringDataNotice';
```
Comment (eugene-manuilov) [inline assets/js/components/GatheringDataNotice.stories.js:27]: Let's use shorter names:
```suggestion
DefaultGatheringDataNotice.storyName = 'Default';
```

File: assets/js/components/GatheringDataNotice.stories.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+export const GatheringDataNoticeOverlay = Template.bind( {} );
+GatheringDataNoticeOverlay.storyName = 'GatheringDataNotice - with overlay';
```
Comment (eugene-manuilov) [inline assets/js/components/GatheringDataNotice.stories.js:30]:
```suggestion
GatheringDataNoticeOverlay.storyName = 'Overlay';
```

File: assets/js/components/GatheringDataNotice.stories.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+GatheringDataNoticeOverlay.args = {
+	style: 'overlay',
+};
+
+export const GatheringDataNoticeSmall = Template.bind( {} );
+GatheringDataNoticeSmall.storyName = 'GatheringDataNotice - small';
```
Comment (eugene-manuilov) [inline assets/js/components/GatheringDataNotice.stories.js:36]:
```suggestion
GatheringDataNoticeSmall.storyName = 'Small';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+
+		dispatch( MODULES_ANALYTICS ).finishResolution( 'getReport', [
+			options,
+		] );
+	},
+};
+
+export const GatheringDataZeroDataStates = Template.bind( {} );
+GatheringDataZeroDataStates.storyName =
+	'Gathering Data with zeroDataStates feature flag enabled';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:187]:
```suggestion
	'Gathering Data w/ zeroDataStates';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+		}
+	},
+};
+GatheringDataZeroDataStates.parameters = {
+	features: [ 'zeroDataStates' ],
+};
+
+export const GatheringDataZeroDataStatesDisabled = Template.bind( {} );
+GatheringDataZeroDataStatesDisabled.storyName =
+	'Gathering Data with zeroDataStates feature flag disabled';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:208]:
```suggestion
	'Gathering Data w/o zeroDataStates';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+		);
+
+		for ( const options of reportOptions ) {
+			dispatch( MODULES_ANALYTICS ).receiveGetReport( {}, { options } );
+		}
+	},
+};
+
+export const LoadedEntityURL = Template.bind( {} );
+LoadedEntityURL.storyName = 'Ready with entity URL set';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:225]:
```suggestion
LoadedEntityURL.storyName = 'Ready w/ entity URL';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+		} );
+
+		for ( const options of reportOptionsWithEntity ) {
+			provideAnalyticsMockReport( registry, options );
+		}
+	},
+};
+
+export const LoadingEntityURL = Template.bind( {} );
+LoadingEntityURL.storyName = 'Loading with entity URL set';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:241]:
```suggestion
LoadingEntityURL.storyName = 'Loading w/ entity URL';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+		} );
+
+		registry
+			.dispatch( MODULES_ANALYTICS )
+			.startResolution( 'getReport', [ reportOptionsWithEntity[ 0 ] ] );
+	},
+};
+
+export const DataUnavailableEntityURL = Template.bind( {} );
+DataUnavailableEntityURL.storyName = 'Data Unavailable with entity URL set';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:257]:
```suggestion
DataUnavailableEntityURL.storyName = 'Data Unavailable w/ entity URL';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+		} );
+
+		registry
+			.dispatch( MODULES_ANALYTICS )
+			.receiveGetReport( [], { options: reportOptionsWithEntity[ 0 ] } );
+	},
+};
+
+export const ZeroDataEntityURL = Template.bind( {} );
+ZeroDataEntityURL.storyName = 'Zero Data with entity URL set';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:276]:
```suggestion
ZeroDataEntityURL.storyName = 'Zero Data w/ entity URL';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+				.dispatch( MODULES_ANALYTICS )
+				.receiveGetReport( zeroing( report ), {
+					options,
+				} );
+		}
+	},
+};
+
+export const ErrorEntityURL = Template.bind( {} );
+ErrorEntityURL.storyName = 'Error with entity URL set';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:298]:
```suggestion
ErrorEntityURL.storyName = 'Error w/ entity URL';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 12]
```diff
@@ -0,0 +1,390 @@
...
+			.receiveError( error, 'getReport', [ options ] );
+		registry
+			.dispatch( MODULES_ANALYTICS )
+			.finishResolution( 'getReport', [ options ] );
+	},
+};
+
+export const GatheringDataEntityURLZeroDataStates = Template.bind( {} );
+GatheringDataEntityURLZeroDataStates.storyName =
+	'Gathering Data with entity URL set and zeroDataStates feature flag enabled';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:325]:
```suggestion
	'Gathering w/ entity URL and zeroDataStates';
```

File: assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js [hunk 13]
```diff
@@ -0,0 +1,390 @@
...
+		}
+	},
+};
+GatheringDataEntityURLZeroDataStates.parameters = {
+	features: [ 'zeroDataStates' ],
+};
+
+export const GatheringDataEntityURLZeroDataStatesDisabled = Template.bind( {} );
+GatheringDataEntityURLZeroDataStatesDisabled.storyName =
+	'Gathering Data with entity URL set and zeroDataStates feature flag disabled';
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardOverallPageMetricsWidget.stories.js:346]:
```suggestion
	'Gathering w/ entity URL and w/o zeroDataStates';
```

File: assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.stories.js [hunk 14]
```diff
@@ -247,6 +247,25 @@ ErrorAnalytics.args = {
 	},
 };
 
+export const GatheringDataZeroDataStates = Template.bind( {} );
+GatheringDataZeroDataStates.storyName =
+	'Gathering Data with zeroDataStates feature flag enabled';
```
Comment (eugene-manuilov) [inline assets/js/modules/search-console/components/dashboard/SearchFunnelWidget/index.stories.js:252]:
```suggestion
	'Gathering w/ zeroDataStates';
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Nice work, @asvinb. Overall looks good to me. Added some suggestion to make story names shorter. Please, take a look.

**Comment by asvinb** [comment]
Thanks @eugene-manuilov ! Definitely the story names are much better with your suggestions! 😁

**Review by eugene-manuilov — APPROVED** [review]
LGTM. Thanks, @asvinb.

**Pull request merged**